src/main/java/org/radarcns/management/service/SourceService.java

```diff
@@ -169,10 +169,40 @@ public void delete(Long id) {
         sourceRepository.delete(id);
     }
 
+    /**
+     * Returns all sources by project in {@link SourceDTO} format
+     * @param projectId
+     * @return list of sources
+     */
     public List<SourceDTO> findAllByProjectId(Long projectId) {
         return sourceRepository.findAllSourcesByProjectId(projectId)
             .stream()
             .map(sourceMapper::sourceToSourceDTO)
             .collect(Collectors.toCollection(LinkedList::new));
     }
+
+    /**
+     * Returns all sources by project in {@link MinimalSourceDetailsDTO} format
+     * @param projectId
+     * @return list of sources
+     */
+    public List<MinimalSourceDetailsDTO> findAllMinimalSourceDetailsByProject(Long projectId) {
+        return sourceRepository.findAllSourcesByProjectId(projectId)
+            .stream()
+            .map(sourceMapper::sourceToMinimalSourceDetailsDTO)
+            .collect(Collectors.toCollection(LinkedList::new));
```

Comment:
Is there any reason to use LinkedList everywhere? It's only good for doing insertions and deletions and even then it's probably better to use LinkedHashSet. Why not just `collect(Collectors.toList())`?

Reply:
@strangepleasures cannot think of a reason why jhipster chose `LinkedList` over a `List`.  As we have similar pattern in all services, we will take this a separate improvement issue.